llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1416,14 +1466,14 @@ void VPlanTransforms::addActiveLaneMask(
   auto *FoundWidenCanonicalIVUser =
       find_if(Plan.getCanonicalIV()->users(),
               [](VPUser *U) { return isa<VPWidenCanonicalIVRecipe>(U); });
-  assert(FoundWidenCanonicalIVUser &&
+  assert(FoundWidenCanonicalIVUser && *FoundWidenCanonicalIVUser &&
```

Comment:
This looks a little odd. Doesn't `find_if` return an iterator? 
```suggestion
  auto IVUsers = Plan.getCanonicalIV()->users();
  /// ...
  assert(FoundWidenCanonicalIVUser != IVUsers.end() && "Must have widened canonical IV when tail folding!"); 
```

Reply:
Done, thanks.